### DISCUSSION TITLE
Getting a the trait " `WorldQuery` is not implemented for `enemy::Enemy` " when trying to query

### DISCUSSION THREAD

**Discussion by Quan1umMango** [bevyengine/bevy#9059] [Q&A] [answered]
Hey so i'm trying to use bevy made a small prototype of the enemy class. it worked fine until i decided to split the Enemy struct code into separate files. There were no major changes done to the code, just add the pub keyword where necessary and importing the stuff. 
But after I finish making them into modules, the code wouldn't compile and give this error:

```
error[E0277]: the trait bound `enemy::Enemy: WorldQuery` is not satisfied
   --> src\enemy\mod.rs:131:17
    |
131 |     mut enemy_q:Query<(Entity,Enemy)>
    |                 ^^^^^^^^^^^^^^^^^^^^^ the trait `WorldQuery` is not implemented for `enemy::Enemy`
    |
    = help: the following other types implement trait `WorldQuery`:
              &'__w mut T
              &T
              ()
              (F0, F1)
              (F0, F1, F2)
              (F0, F1, F2, F3)
              (F0, F1, F2, F3, F4)
              (F0, F1, F2, F3, F4, F5)
            and 55 others
```
I tried to revert the changes but that didn't seem to work as well. What am i doing wrong?
Here is a view of the Enemy struct:
```rs
#[derive(Component, Clone)]
pub struct Enemy {
     id: Uuid,
     position: Vec3,
     color: Color,
     size: Vec2,
     health: u32,
     speed: f32,
     dir: Vec3,
     goto: Vec3,
     point_no: usize,
}
```
I googled the error and got [this](https://github.com/bevyengine/bevy/issues/2616), but it didnt seem to help.

Bevy Version: 0.10.1
Platform: Windows 10

**Comment by MrGVSV** [comment, marked as answer]
It should be `mut enemy_q:Query<(Entity,&Enemy)>` (note the `&` before `Enemy`, also note that `Entity` is the only type to not require a `&`).

**Reply by Quan1umMango** [reply]
Wow I completely missed that. Thanks

**Reply by MrGVSV** [reply]
No worries! It's pretty easy to miss 😄
